src/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -28,92 +31,143 @@ public ToolManifestFinder(DirectoryPath probStart, IFileSystem fileSystem = null
 
         public IReadOnlyCollection<ToolManifestPackage> Find(FilePath? filePath = null)
         {
-            var result = new List<ToolManifestPackage>();
-
             IEnumerable<FilePath> allPossibleManifests =
                 filePath != null
-                    ? new[] { filePath.Value }
+                    ? new[] {filePath.Value}
                     : EnumerateDefaultAllPossibleManifests();
 
+            bool findAnyManifest = false;
+            var result = new List<ToolManifestPackage>();
             foreach (FilePath possibleManifest in allPossibleManifests)
             {
                 if (_fileSystem.File.Exists(possibleManifest.Value))
                 {
-                    SerializableLocalToolsManifest deserializedManifest = JsonConvert.DeserializeObject<SerializableLocalToolsManifest>(
```

Comment:
Please use split screen diff for this change. it is a extract method refactoring, the change looks big

Reply:
I should create a separate commit for extract method...

only the following is new production code

```c#
 foreach (ToolManifestPackage p in GetToolManifestPackageFromOneManifestFile(deserializedManifest, possibleManifest))
                    {
                        if (!result.Any(addedToolManifestPackages =>
                            addedToolManifestPackages.PackageId.Equals(p.PackageId)))
                        {
                            result.Add(p);
                        }
                    }
                     if (deserializedManifest.isRoot)
                    {
                        return result;
                    }
```